test_util/dcos_api_session.py

```diff
@@ -89,31 +68,92 @@ def api_request(self, method, path_extension, *, scheme=None, host=None, query=N
 
 
 class DcosApiSession(ARNodeApiClientMixin, RetryCommonHttpErrorsMixin, ApiClientSession):
-    def __init__(self, dcos_url: str, masters: list, public_masters: list,
-                 slaves: list, public_slaves: list, default_os_user: str,
-                 auth_user: Optional[DcosUser]):
-        """Proxy class for DC/OS clusters.
+    def __init__(
+            self,
+            dcos_url: str,
+            masters: Optional[List[str]],
+            slaves: Optional[List[str]],
+            public_slaves: Optional[List[str]],
+            default_os_user: str,
+            auth_user: Optional[DcosUser]):
+        """Proxy class for DC/OS clusters. If any of the host lists (masters,
+        slaves, public_slaves) are provided, the wait_for_dcos function of this
+        class will wait until provisioning is complete. If these lists are not
+        provided, then there is no ground truth and the cluster will be assumed
+        the be in a completed state.
 
         Args:
             dcos_url: address for the DC/OS web UI.
             masters: list of Mesos master advertised IP addresses.
-            public_masters: list of Mesos master IP addresses routable from
-                the local host.
             slaves: list of Mesos slave/agent advertised IP addresses.
+            public_slaves: list of public Mesos slave/agent advertised IP addresses.
             default_os_user: default user that marathon/metronome will launch tasks under
             auth_user: use this user's auth for all requests
                 Note: user must be authenticated explicitly or call self.wait_for_dcos()
         """
         super().__init__(Url.from_string(dcos_url))
-        self.masters = sorted(masters)
-        self.public_masters = sorted(public_masters)
-        self.slaves = sorted(slaves)
-        self.public_slaves = sorted(public_slaves)
-        self.all_slaves = sorted(slaves + public_slaves)
+        self.master_list = masters
+        self.slave_list = slaves
+        self.public_slave_list = public_slaves
```

Comment:
Nvm, I see the `sorted` on the getters below now. 

Though, I do wonder why we don't just sort them to begin with and return that?